src/fondant/pipeline/lightweight_component.py

```diff
@@ -40,3 +50,47 @@ def image(cls) -> Image:
         return AppliedPythonComponent
 
     return wrapper
+
+
+def build_python_script(component_cls: t.Type[Component]) -> str:
+    """Build a self-contained python script for the provided component class, which will act as
+    the `src/main.py` script to execute the component.
+    """
+    imports_source = textwrap.dedent(
+        """\
+        from typing import *
```

Comment:
Nitpick: If this might get bigger in the future, maybe it would make sense to extract it to a separate file for readability.